app/models/batch_process.rb

```diff
@@ -89,4 +89,38 @@ def refresh_metadata_cloud
       refresh_metadata_cloud_mets
     end
   end
+
+  def batch_status
+    if single_status
+      single_status
+    elsif statuses[:failed] != 0
+      percent = ((statuses[:failed] / statuses[:total]) * 100).round
+      "#{percent}% of parent objects have a failure."
+    else
+      "Batch status unknown"
+    end
+  end
+
+  def single_status
+    if statuses[:in_progress] / statuses[:total] == 1
```

Comment:
I think you might want to capture the statuses one time here - I think the way you have this structured, you might be calling batch_connections.map up to 6 times.

```
current_status = statuses
if current_status[:in_progress] / current_status[:total] == 1
... etc.
```